gsw/tests/test_check_functions.py

```diff
@@ -43,10 +43,12 @@ def cfcf(request):
 def test_check_function(cfcf):
     cv, cf, mfunc = cfcf
     mfunc.run(locals())
-    if mfunc.exception is not None:
+    if mfunc.exception is not None or not mfunc.passed:
         print('\n', mfunc.name)
         print('  ', mfunc.runline)
         print('  ', mfunc.testline)
-        print("   >>>%s<<<" % mfunc.exception)
+        if mfunc.exception is None:
+            mfunc.exception = ValueError('Calculated values are different from the expected matlab results.')
```

Comment:
Can you output the values here to make this message clearer? (I am a little bit lost in the auto-generated test.)

BTW we have some values that are mismatching due to the algorithm difference.
But we also have some NaNs in the putput that are not present in the comparison values.
I believe the NaNs we are getting are correct because the are present in the input `cv.p_chck_cast` and should be in the output `cv.distance`.